betting-game-vue/src/store/modules/matchday.store.ts

```diff
@@ -0,0 +1,52 @@
+import { createModule, mutation, action, getter } from 'vuex-class-component'
+import Matchday from '@/models/Matchday'
+import http from '@/utils/Http'
+
+type MatchdayProps = { year: string; matchday: string }
+
+const VuexModule = createModule({
+  strict: false
+})
+
+export class MatchdayStore extends VuexModule {
+  @getter data?: Matchday
+  @getter isLoading: boolean = true
+  @getter hasError: boolean = false
+  @getter errorMsg?: string
+
+  @mutation FETCH_DATA_PENDING(): void {
```

Comment:
I'm curious, is there any reason why the mutation names are all caps?

Reply:
By convention, constants are used for mutation types ([https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types](https://vuex.vuejs.org/guide/mutations.html#using-constants-for-mutation-types)). This was the reason why I made them upper case. 🤷‍♂️

Reply:
okay, but if you're just declaring a normal mutation in the state, camel case would make more sense. I'm not really against using constants but if we're using it, better use it correctly by declaring in a 3rd party .ts file like the example, so that we can declare different mutation names across stores. Should also be applicable for actions.
```
// mutation-types.js
export const SOME_MUTATION = 'SOME_MUTATION'
```

Reply:
Okay, I changed it to camel case 👍